py3status/modules/gcal.py

```diff
@@ -0,0 +1,160 @@
+# -*- coding: utf-8 -*-
+"""
+Display upcoming calendar events from Google Calendar.
+
+Configuration parameters:
+    agenda_days: show calendar events for number of days (default 1)
+    button_refresh: mouse button to refresh this module (default 2)
+    cache_timeout: refresh interval for this module (default 3600)
+    command: customize gcalcli to dictate what to do with the events
+        (default 'gcalcli agenda --nocolor')
+    format: display format for this module (default '{format_event}')
+    format_event: display format for events
+        (default '[\?if=is_date&color=#ff0 %a %b %d ]
+        [\?if=is_time&color=#0f0 %-I:%M %p ][\?color=#f00 {event}]')
+    format_separator: show separator only if more than one (default ', ')
+    remove: list of strings to remove from the output (default [])
+
+Format placeholders:
+    {format_event} format for events
+
+format_event placeholders:
+    {event} scheduled event eg Volleyball
+    %?  strftime characters to be translated. See `man strftime`.
+    is_time: a boolean based on event containing a time schedule
+    is_date: a boolean based on event containing a date schedule
+
+Requires:
+    gcalcli: Google Calendar CLI
+
+@author lasers
+
+SAMPLE OUTPUT
+[
+    {'color': '#FFFF00', 'full_text': u'Thu Jul 20 '},
+    {'color': '#FF0000', 'full_text': u'Tacos'},
+    {'full_text': u', '},
+    {'color': '#FF0000', 'full_text': u'Volleyball'},
+    {'full_text': u', '},
+    {'color': '#00FF00', 'full_text': u'6:00 AM '},
+    {'color': '#FF0000', 'full_text': u'Garbage Pickup'},
+    {'full_text': u', '}
+    {'color': '#FFFF00', 'full_text': u'Sat Jul 22 '},
+    {'color': '#00FF00', 'full_text': u'6:05 PM '},
+    {'color': '#FF0000', 'full_text': u'Flight to Poland'}
+]
+"""
+
+from datetime import datetime, timedelta
+import re
+
+STRING_NOT_INSTALLED = "isn't installed"
+FORMAT_GCAL = '%Y%m%d'
+FORMAT_DATE = '%a %b %d'
+FORMAT_TIME = '%I:%M%p'
```

Comment:
These are still confusingly named as this is the format for calling gcalcli rather than outputting from the module.

Clarity is good for all the family.